Packs/AzureDevOps/Integrations/AzureDevOps/AzureDevOps_test.py

```diff
@@ -822,3 +825,1006 @@ def test_generate_login_url(mocker):
+
+    # setting parameters
+    project = 'test'
+    repository = 'xsoar'
+    reviewer_user_id = 'testestest6565'
+
+    url = f'https://dev.azure.com/{ORGANIZATION}/{project}/_apis/git/repositories/{repository}/' \
+          f'pullRequests/{pull_request_id}/reviewers/{reviewer_user_id}'
+
+    mock_response = json.loads(load_mock_response(mock_response_path))
+    requests_mock.put(url, json=mock_response)
+
+    client = Client(
+        client_id=CLIENT_ID,
+        organization=ORGANIZATION,
+        verify=False,
+        proxy=False,
+        auth_type='Device Code')
+
+    result = pull_request_reviewer_add_command(client, {'pull_request_id': pull_request_id, 'reviewer_user_id': reviewer_user_id},
+                                               ORGANIZATION, repository, project)
+    if pull_request_id == '42':
+        # PR does not exist
+        assert 'The requested pull request was not found.' in result.outputs.get("message")
+    else:
+        assert result.outputs_prefix == 'AzureDevOps.PullRequestReviewer'
+        assert result.readable_output == 'TEST (TEST) was created successfully as a reviewer for Pull Request ID 40.'
+
+
+@pytest.mark.parametrize('pull_request_id, mock_response_path',
+                         [('40', 'pull_request_commit_list.json'),
+                          ('42', 'pull_request_not_found.json')])
+def test_pull_request_commit_list_command(requests_mock, pull_request_id: str, mock_response_path: str):
+    """
+    Given:
+     - all arguments
+    When:
+     - executing azure-devops-pull-request-commit-list command
+    Then:
+     - Ensure outputs_prefix and readable_output are set up right
+     - Ensure informative message when pull_request_id does not exist
+    """
+    from AzureDevOps import Client, pull_request_commit_list_command
+
+    authorization_url = 'https://login.microsoftonline.com/organizations/oauth2/v2.0/token'
+    requests_mock.post(authorization_url, json=get_azure_access_token_mock())
+
+    # setting parameters
+    project = 'test'
+    repository = 'xsoar'
+
+    url = f'https://dev.azure.com/{ORGANIZATION}/{project}/_apis/git/repositories/{repository}/' \
+          f'pullRequests/{pull_request_id}/commits'
+
+    mock_response = json.loads(load_mock_response(mock_response_path))
+    requests_mock.get(url, json=mock_response)
+
+    client = Client(
+        client_id=CLIENT_ID,
+        organization=ORGANIZATION,
+        verify=False,
+        proxy=False,
+        auth_type='Device Code')
+
+    result = pull_request_commit_list_command(client, {'pull_request_id': pull_request_id, 'limit': '1'},
+                                              ORGANIZATION, repository, project)
+    if pull_request_id == '42':
+        # PR does not exist
+        assert 'The requested pull request was not found.' in result.raw_response.get("message")
+    else:
+        assert result.readable_output.startswith('### Commits')
```

Comment:
compare the `output` to expected outputs

Reply:
fixed